### PR TITLE
MGMT-6534 - Fix PEP8 violations on tools and add type hints annotations

### PR DESCRIPTION
**Pull request by eliorerz**

### CODE REVIEW

File: discovery-infra/test_infra/tools/terraform_utils.py [hunk 1]
```diff
@@ -36,12 +36,12 @@ def change_variables(self, variables, refresh=True):
             json.dump(tfvars, _file)
         self.apply(refresh=refresh)
 
-    def get_state(self):
+    def get_state(self) -> None:
```
Comment (YuviGold) [inline discovery-infra/test_infra/tools/terraform_utils.py:39]: This doesn't look right

File: discovery-infra/test_infra/tools/concurrently.py [hunk 2]
```diff
@@ -15,13 +16,17 @@ def _safe_run(job, job_id, done_handler):
             done_handler(job_id)
 
 
-def run_concurrently(jobs, done_handler=None, max_workers=5, timeout=2 ** 31):
+def run_concurrently(
+    jobs: Union[List, Dict, Tuple],
+    done_handler: Callable[[int], None] = None,
+    max_workers: int = 5,
+    timeout: float = 2 ** 31,
+) -> dict:
```
Comment (YuviGold) [inline discovery-infra/test_infra/tools/concurrently.py:24]: What `dict`? Please use `Dict[KT, VT]` notation

File: discovery-infra/test_infra/tools/concurrently.py [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import logging
 from concurrent.futures import ThreadPoolExecutor
+from typing import Callable, Dict, List, Tuple, Union
 
 
-def _safe_run(job, job_id, done_handler):
+def _safe_run(job, job_id: int, done_handler: Callable[[int], None]):
```
Comment (YuviGold) [inline discovery-infra/test_infra/tools/concurrently.py:6]: Use `Optional`

Comment (eliorerz) [reply discovery-infra/test_infra/tools/concurrently.py:6]: Where? `Callable[[int], None]` annotate the function type so done handler signature must be `handler(x: int) -> None`

**Review by YuviGold — APPROVED** [review]

**Pull request merged**